docs: update README to reflect supported maintenance version

juliaofferman opened the pull request:
MySQL vector syntax has changed in the latest maintenance version (see https://cloud.google.com/sql/docs/mysql/work-with-vectors). We are yet to update the integration to use this new syntax, so need to call out the supported versions while we work on a new version of the integration. The updated syntax includes changing the ANN search function name, parameters, and where it can appear in a query. Additionally, create/drop/alter vector index syntax has changed. There are also some new restrictions, ie a table must have 1000 rows before creating an index.

juliaofferman committed:
Update doc to reflect supported maintenance version

MySQL vector syntax has changed in the latest maintenance version (see https://cloud.google.com/sql/docs/mysql/work-with-vectors). We are yet to update the integration to use this new syntax, so need to call out the supported versions while we work on a new version of the integration.

jackwotherspoon commented:
/gcbrun

jackwotherspoon commented:
@juliaofferman do you mind adding to the PR description some details highlighting what the syntax changes are?

averikitsch reviewed (approved):
Approving, but can you also add this to the README and https://github.com/googleapis/langchain-google-cloud-sql-mysql-python/blob/main/docs/vector_store.ipynb? Thanks!

averikitsch committed:
Merge branch 'main' into patch-1

averikitsch commented:
/gcbrun

jackwotherspoon reviewed (approved):
LGTM, ditto Averi's comment of adding a note to README.

Also see my other comment about updating description to give more details into what the differences in syntax are now.

juliaofferman committed:
Add supported maintenance version to README & notebook

Add note that maintenance versions between
MYSQL_8_0_36.R20240401.03_00 and
MYSQL_8_0_36.R20241208.01_00 are required to the
overall README and vector_store notebook.

jackwotherspoon commented:
/gcbrun

juliaofferman committed:
Fix underline in maintenance version heading

Underline was shorter than the section heading, causing "sphinx.errors.SphinxWarning: /home/runner/work/langchain-google-cloud-sql-mysql-python/langchain-google-cloud-sql-mysql-python/docs/README.rst:35:Title underline too short.". Fixed the underline to match text

juliaofferman commented:
/gcbrun

jackwotherspoon commented:
/gcbrun
